### ISSUE TITLE
Add New Loss

### ISSUE THREAD

**Issue by interestingzhuo** [KevinMusgrave/pytorch-metric-learning#589]
I implemented the PNP Loss. How to add it in this repo?  (Zhuo Li et al. “Rethinking the Optimization of Average Precision: Only Penalizing Negative Instances before Positive Ones is Enough” Proceedings of the ... AAAI Conference on Artificial Intelligence(2021): n. pag.)

**Comment by KevinMusgrave** [comment]
Thanks for your interest!

Please fork this repo, checkout the dev branch, and make a pull request with your changes.

The loss function should extend BaseMetricLossFunction:

```python
from ..losses import BaseMetricLossFunction

class PNPLoss(BaseMetricLossFunction):
    def compute_loss(self, embeddings, labels, indices_tuple, ref_emb, ref_labels):
        # get PNP loss
        ...

        # put into dictionary #
        return {
            "loss": {
                "losses": pnp_loss,
                "indices": None,
                "reduction_type": "already_reduced",
            }
        }
```

For now, just use `embeddings` and `labels` and ignore the other inputs. Then open a pull request and we can discuss further.

**Comment by KevinMusgrave** [comment]
Available in v2.1.0
